processing/src/main/java/org/apache/druid/segment/join/PostJoinCursor.java

```diff
@@ -70,6 +70,15 @@ public static PostJoinCursor wrap(
   }
 
   private void advanceToMatch()
+  {
+    if (valueMatcher != null) {
+      while (!isDone() && !valueMatcher.matches(false)) {
+        baseCursor.advance();
+      }
+    }
+  }
+
+  private void advanceToMatchUninterruptibly()
```

Comment:
Please leave a comment here that this can be a long-running CPU call. Which is why advanceUninterruptibly is not directly used in advance() call unlike other cursors. Please link the github issue in the comment.

Reply:
Done